substrate/frame/revive/src/vm/evm.rs

```diff
@@ -119,6 +120,33 @@ fn run<WIRE: InterpreterTypes>(
 	}
 }
 
+/// Re-implementation of REVM run_plain function to add trace logging to our EVM interpreter loop.
+fn run_plain<WIRE: InterpreterTypes>(
+	interpreter: &mut Interpreter<WIRE>,
+	instruction_table: &revm::interpreter::InstructionTable<WIRE, DummyHost>,
+	host: &mut DummyHost,
+) -> InterpreterAction {
+	use crate::{alloc::string::ToString, format};
+	use revm::bytecode::OpCode;
+	while interpreter.bytecode.is_not_end() {
+		log::trace!(
+			"[{}]: {}, stacktop: {}, memory size: {} {:?}",
+			interpreter.bytecode.pc(),
+			OpCode::new(interpreter.bytecode.opcode())
+				.map_or("INVALID".to_string(), |x| format!("{:?}", x.info())),
+			interpreter.stack.top().map_or("None".to_string(), |x| format!("{:#x}", x)),
```

Comment:
Thanks, looks good. Is there a way to print more than just the top of the stack? Many functions pop more than one argument. 5 values would be a good limit I think